File: JavaSpringBackend/src/main/java/com/api/carpintech/models/Material.java
package com.api.carpintech.models;

import jakarta.persistence.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.UUID;

@Entity
@Table(name = "material")
public class Material {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @Column(name = "nome", nullable = false, length = 100)
    private String nome;

    @Column(name = "custo", nullable = false)
    private Double custo;

    @ManyToOne
    @JoinColumn(name = "estoque_id", nullable = false)
    private Estoque estoque;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Double getCusto() {
        return custo;
    }

    public void setCusto(Double custo) {
        this.custo = custo;
    }
    

}

File: MarcenariaApi/Models/Material.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;

namespace MarcenariaApi.Models
{
    public class Material
    {
        [Key]
        public int? id { get; set; }
        public string? nome { get; set; }
        public Double custo { get; set; }
        public int estoqueId { get; set; }

        [JsonIgnore]
        public Estoque? estoque;
    }
}
